### DISCUSSION TITLE
tsml_front_page home page meeting search, not river of news

### DISCUSSION THREAD

**Discussion by douggreen** [code4recovery/12-step-meeting-list#832] [Q&A] [answered]
I'd like my sites home page to be the meeting list (search) page. 

This site is currently in  development so the URL is https://dev.blueridgeareaintergroup.org

I've changed the home page using 

```
// Make the meeting guide the home page.
// @ee https://wordpress.org/plugins/12-step-meeting-list/
add_action('pre_get_posts', 'tsml_front_page');
```

But it shows a "River of News", that is, it shows my two meetings listed one after the other.

I'd like it to instead be https://dev.blueridgeareaintergroup.org/meetings/

**Comment by joshreisner** [comment, marked as answer]
hm, it doesn't appear this technique works anymore, we should probably create an issue to remove it.

try this: if you don't have a child theme already create one, and add a page called `front-page.php` and add this to it:

```
<?php

$file = !empty($tsml_user_interface) && $tsml_user_interface === 'tsml_ui' ? 'tsml-ui' : 'meetings';

$file = WP_PLUGIN_DIR . '/12-step-meeting-list/templates/archive-' . $file . '.php';

if (file_exists($file)) {
    include($file);
}
```

**Reply by douggreen** [reply]
That works, but it breaks my child theme include of css. I've left the site with this, so that you can see the results.

front-page.php:
```
<?php

// Make the meeting guide the home page.
// @see https://github.com/code4recovery/12-step-meeting-list/discussions/832
$file = !empty($tsml_user_interface) && $tsml_user_interface === 'tsml_ui' ? 'tsml-ui' : 'meetings';
$path = WP_PLUGIN_DIR . '/12-step-meeting-list/templates/archive-' . $file . '.php';
if (file_exists($path)) {
  @require_once $path;
}
```

functions.php:
```
<?php
/**
 * Twenty Twenty BRI functions and definitions
 */

/**
 * Register the styles.
 */
function twentytwenty_bri_register_styles() {
  $theme = wp_get_theme();
  wp_enqueue_style('twentytwenty-style', get_template_directory_uri() . '/style.css', [], $theme->parent()->get('Version'));
  wp_enqueue_style('bri-style', get_stylesheet_uri(), [], $theme->get('Version'));
}

add_action('wp_enqueue_scripts', 'twentytwenty_bri_register_styles');
```

**Reply by joshreisner** [reply]
hm, strange. if i install twenty twenty, make a child theme using the [child theme configurator](https://wordpress.org/plugins/child-theme-configurator/) plugin, add that file, and load the example CSV then this is what I see: 
![Screenshot 2022-06-19 at 14-57-50 wordpress demo – Just another WordPress site](https://user-images.githubusercontent.com/1551689/174501942-ff54a6ef-73fe-4166-93f9-3112c030bc0f.png)

not sure what happened there, but maybe try recreating the child theme?

**Reply by douggreen** [reply]
I've compared the CSS. When I add the snippet above, I'm missing  https://dev.blueridgeareaintergroup.org/wp-content/plugins/12-step-meeting-list/assets/css/public.min.css?ver=3.14.4

**Reply by douggreen** [reply]
I thought adding ```tsml_assets();``` to front-page.php would work, but it didn't.

**Reply by douggreen** [reply]
It also needs
```
global $post_type;
$post_type = 'tsml_meeting';
tsml_assets();
```

**Reply by joshreisner** [reply]
That's great you got it fixed. Not sure why your site needs those additional lines, but happy you were able to fix it.

**Reply by joshreisner** [reply]
marking this question as "answered" -- i've [created an issue](https://github.com/code4recovery/12-step-meeting-list/issues/841) based on this discussion
